benchmarks/bench_ops.cpp

```diff
@@ -0,0 +1,137 @@
+#include <complex>
+#include <functional>
+#include <type_traits>
+
+#include <sycl/sycl.hpp>
+
+#include "sycl_ext_complex.hpp"
+
+#include <oneapi/dpl/random>
+
+#include <benchmark/benchmark.h>
+
+enum class OpName { PLUS, MINUS, MULTIPLIES, DIVIDES };
+
+template <typename T, OpName opname> struct Op;
+
+template <typename T> struct Op<T, OpName::PLUS> {
+  T operator()(const T &a, const T &b) const { return a + b; }
+};
+
+template <typename T> struct Op<T, OpName::MINUS> {
+  T operator()(const T &a, const T &b) const { return a - b; }
+};
+
+template <typename T> struct Op<T, OpName::MULTIPLIES> {
+  T operator()(const T &a, const T &b) const { return a * b; }
+};
+
+template <typename T> struct Op<T, OpName::DIVIDES> {
+  T operator()(const T &a, const T &b) const { return a / b; }
+};
+template <bool Ext, typename R, OpName opname, std::uint32_t SEED = 777>
+static void BM_binary_op(benchmark::State &state) {
+  sycl::queue Q(sycl::gpu_selector_v);
+
+  using T = typename std::conditional_t<Ext, sycl::ext::cplx::complex<R>,
+                                        std::complex<R>>;
+  using OpClass = Op<T, opname>;
+
+  int n = state.range(0);
+
+  auto a = sycl::malloc_device<T>(n, Q);
+  auto b = sycl::malloc_device<T>(n, Q);
+  auto c = sycl::malloc_device<T>(n, Q);
+
+  OpClass op{};
+
+  Q.submit([&](sycl::handler &CGH) {
+     CGH.parallel_for(sycl::range<1>(n), [=](sycl::item<1> idx) {
```

Comment:
I suggest using the Q.parrallel_for if we don't need the CGH except for launching the kernel

Reply:
fixed.